src/shogun/io/Serializable.h

```diff
@@ -127,31 +127,51 @@ template<class T> class CMatrixSerializable: public CSerializable<SGMatrix<T>>
 	virtual const char* get_name() const { return "MatrixSerializable"; }
 };
 
-// FIXME: there is no SG_ADD for SGStringList so need to do that manually.
-// can be dropped once SG_ADD works with SGStringList
 // Note: cannot inherit from CSerializable as need to overload/change init()
-template<class T> class CStringListSerializable: public CSGObject
+template<class T> class CStdVectorSerializable: public CSGObject
 {
 public:
-	CStringListSerializable() : CSGObject() { init(); }
-	CStringListSerializable(SGStringList<T> value, const char* value_name="") : CSGObject()
+	CStdVectorSerializable() : CSGObject() { init(); }
+	CStdVectorSerializable(const std::vector<T>& value, const char* value_name="") 
+		: CSGObject(), m_name(value_name)
 	{
 		init();
 		m_value = value;
 	}
-	virtual ~CStringListSerializable() {}
-	virtual const char* get_name() const { return "StringListSerializable"; }
+	virtual ~CStdVectorSerializable() {}
+	virtual const char* get_name() const { return "StdVectorSerializable"; }
 
 protected:
 	virtual void init()
 	{
-		set_generic<T>();
-		m_value = SGStringList<T>();
-		m_parameters->add_vector(&m_value.strings, &m_value.num_strings, "value",
-				"Serialized value");
+		if constexpr (type_internal::is_sg_primitive<T>::value)
+			set_generic<T>();
+		// FIXME(?): std::vector<bool> is not a container
+		// doesn't play well with the parameter framework
+		if constexpr (!std::is_same<T, bool>::value)
+			watch_param("value", &m_value);
+		else
+			SG_ERROR("std::vector<bool> is not currently serializable. "
```

Comment:
shouldn't this be a compile time error?

Reply:
Yes, but class_list instantiates it for all types, so if I did that it wouldn't compile

Reply:
ah i see, nvm then